sample_consensus/include/pcl/sample_consensus/impl/sac_model_registration.hpp

```diff
@@ -83,7 +83,16 @@ pcl::SampleConsensusModelRegistration<PointT>::computeModelCoefficients (const I
   Indices indices_tgt (3);
   for (int i = 0; i < 3; ++i)
   {
-    indices_tgt[i] = correspondences_.at (samples[i]);
+    try
```

Comment:
This try-catch is same as using `correspondences_.find` with a check for `correspondences.end()`.

Reply:
Yes, that would be possible, too. IMO this try-catch is a bit nicer since it does not involve an iterator. Unless you see a reason to use `find`, I will leave it as is.

Reply:
Before providing reasons, let me ack that the situation we're trying to detect is exceptional and this is a corner case. My reasons to not use try-catch here might not be convincing, but they are not rooted in just personal preference

1. exception handling is not used widely in PCL (or idomatic C++) for control flow (200 uses in library, most of them in io module)
2. clarity of code: find and check is widely used and thus easier to "maintain"
3. try-catch adds a lot of code for handling (larger binaries)

As such, I feel this is unwarranted use of try-catch